omroneipApp/src/drvOmroneip.cpp

```diff
@@ -0,0 +1,1882 @@
+        }
+        catch(...)
+        {
+          // attempt to split name and integer
+          for (size_t n = 0; n<words.front().size(); n++)
+          {
+            if (words.front().c_str()[n] == '[')
+            {
+              std::string offsetSubstring = words.front().substr(n+1,words.front().size()-(n+1));
+              indexStartPos = n;
+              for (size_t m = 0; m<offsetSubstring.size(); m++)
+              {
+                if (offsetSubstring.c_str()[m] == ']')
+                {
+                  try
+                  {
+                    //struct integer found
+                    structIndex = std::stoi(offsetSubstring.substr(0,m));
+                    keyWords.at("optimisationFlag") = "attempt optimisation";
+                    goto findOffsetFromStruct;
+                  }
+                  catch(...)
+                  {
+                    std::cout << "Error, could not find a valid index for the requested structure: " << words.front() << std::endl;
+                    keyWords.at("stringValid") = "false";
+                  }
+                }
+              }
+            }
+          }
+          std::cout<<"Invalid index for requested structure: " << words.front() << std::endl;
+          keyWords.at("stringValid") = "false";
+          
+          //look for matching structure in structMap_
+          //if found, look for the offset at the structIndex within the structure
+          findOffsetFromStruct:
+            std::string structName = words.front().substr(0,indexStartPos);
+            bool structFound = false;
+            for (auto item: this->structMap_)
+            {
+              if (item.first == structName)
+              {
+                //requested structure found
+                structFound = true;
+                if (structIndex>=item.second.size())
+                {
+                  std::cout<<"Error! Attempt to read index: " << structIndex << " from struct: " << item.first << " failed." << std::endl;
+                  keyWords.at("stringValid") = "false";
+                }
+                else
+                {
+                  offset = item.second[structIndex];
+                  break;
+                }
+              }
+            }
+            if (!structFound)
+            {
+              std::cout << "Could not find structure requested: " << words.front() << ". Have you loaded a struct file?" << std::endl;
+              keyWords.at("stringValid") = "false";
+            }
+        }
+        keyWords.at("offset") = std::to_string(offset);
+      }
+      words.pop_front();
+    }
+    else if (i == 4)
+    {
+      // Check for valid extra attributes
+      /* These attributes overwrite libplctag attributes, other attributes which arent overwritten are not mentioned here
+        Users can overwrite these defaults and other libplctag defaults from their records */
+      std::unordered_map<std::string, std::string> defaultTagAttribs = {
```

Comment:
This `unordered_map<std::string, std::string>` repeats in a lot of places. Could we either typedef it or perhaps wrap it in a class/object?

Reply:
I have implemented:
typedef std::unordered_map<std::string, std::vector<std::string>> structDtypeMap;
typedef std::unordered_map<std::string, std::string> drvInfoMap;

To tidy up the two worst offenders.